tests/testenv/cli_test_env.go

```diff
@@ -242,7 +242,7 @@ func trimOutput(s string) string {
 
 // ListSnapshotsAndExpectSuccess lists given snapshots and parses the output.
 func (e *CLITest) ListSnapshotsAndExpectSuccess(t *testing.T, targets ...string) []SourceInfo {
-	lines := e.RunAndExpectSuccess(t, append([]string{"snapshot", "list", "-l", "--manifest-id"}, targets...)...)
+	lines := e.RunAndExpectSuccess(t, append([]string{"snapshot", "list", "-l", "--manifest-id", "--max-results", "400"}, targets...)...)
```

Comment:
why do we need this?

Reply:
You are right, it's no longer strictly needed. It was added because I originally included `[]string{"false", "true", "inherit"}` for both files and directories in `TestSnapshotFail`. To cut down on test time significantly, I dropped the "inherit" parameter (instead randomly picking "inherit" vs "false" since their behavior is the same) as well as the `0500` permission test cases.

The test checks that the number of expected successful snapshots matches the number reported by `snapshot list` at the end. It was previously failing as the number of snapshots exceeded the `snap list` max size. With these test time improvements, the number of snapshots per source drops to under the default 100 max list size, so yeah this change is no longer strictly necessary.